### ISSUE TITLE
[java] RedundantFieldInitializer: false positive with lombok's @Value

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4458]
**Affects PMD Version:7.0.0-rc1** 

**Rule: RedundantFieldInitializer**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_performance.html#redundantfieldinitializer>

**Description:**
The [@Value](https://projectlombok.org/features/Value) annotation can add a `final` modifier to the field `bar`, so PMD should not report a warning at line 4. 

**Code Sample demonstrating the issue:**

```java
import lombok.Value;
@Value
public class Test {
    String bar = null;   // report a warning
}
```

**Expected outcome:**
PMD reports a violation at line 4, but that's wrong. That's a false positive. 

**Running PMD through:** *CLI*
